### PR TITLE
Handle keyword metavariables during macro expansion

### LINKED ISSUE #2194: Handle ```$type``` macro parameter
I tried this code:

```rust
macro_rules! foo {($type:ident) => {
    let $type = 12;
}} 

pub fn foo() {
    foo!(_a);
}
```

I expected to see this happen: no error

Instead, this happened:

```
test.rs:2:10: error: unexpected token in macro transcribe: expected ‘(’ or identifier after ‘$’, got ‘type’
    2 |     let $type = 12;
      |          ^~~~
```

### Meta

git: ```bb1e40da864addfc2d8dfc310928b7a4e7b13185```

### PR DESCRIPTION
**Pull request by powerboat9**
Fixes #2194

### CODE REVIEW

**Review by philberty — CHANGES_REQUESTED** [review]
Hmmm I think this LGTM but just wondering if get_str_handle_keyword should just replace the get_str function? I think the only reason get_str dodesnt work is jsut because any keyword has a null token but has a token id set and this fixes it. I wonder can you check that?

Otherwise i think the get_str and get_str_handle_keyword just seems confusing.

**Comment by powerboat9** [comment]
Should be fixed now

File: gcc/rust/expand/rust-macro-substitute-ctx.cc [hunk 1]
```diff
@@ -199,8 +200,13 @@ std::pair<std::vector<std::unique_ptr<AST::Token>>, size_t>
 SubstituteCtx::substitute_token (size_t token_idx)
 {
   auto &token = macro.at (token_idx);
+
+  if (token_id_is_keyword (token->get_id ()))
+    goto case_keyword;
```
Comment (philberty) [inline gcc/rust/expand/rust-macro-substitute-ctx.cc:205]: do you need this goto? 

I wonder can you turn the case IDENTIFIER into the default case and leave the case LEFT_PAREN as is?

**Comment by philberty** [comment]
Also thanks for fixed the get_str is looking brilliant

**Comment by powerboat9** [comment]
```goto``` removed

**Review by CohenArthur — APPROVED** [review]
LGTM

**Pull request merged**